### ISSUE TITLE
Cannot override `jwtTokenEnhancer` bean.

### ISSUE THREAD

**Issue by tsachev** [spring-projects/spring-security-oauth2-boot#231]
The bean `jwtTokenEnhancer` defined in `AuthorizationServerTokenServicesConfiguration.JwtTokenServicesConfiguration` cannot be overridden, because its not conditional.

I need to override it because I want to inject `UserDetailsService` so the `Authentication#getPrincipal()` to return not just the username.

My current workaround is 
```
@Configuration
public class TokenConfig
{
    private final AuthorizationServerProperties authorization;

    public TokenConfig(AuthorizationServerProperties authorization)
    {
        this.authorization = authorization;
    }

    @Bean
    public TokenStore jwtTokenStore(UserDetailsService userDetailsService) throws Exception
    {
        return new JwtTokenStore(jwtTokenEnhancer(userDetailsService));
    }

    // we cannot just override @Bean jwtTokenEnhancer from AuthorizationServerTokenServicesConfiguration because it's not conditional.
    private JwtAccessTokenConverter jwtTokenEnhancer(UserDetailsService userDetailsService) throws Exception
    {
        String keyValue = this.authorization.getJwt().getKeyValue();
        Assert.notNull(this.authorization.getJwt().getKeyValue(), "keyValue cannot be null");

        JwtAccessTokenConverter converter = new JwtAccessTokenConverter();
        if (!keyValue.startsWith("-----BEGIN")) {
            converter.setVerifierKey(keyValue);
        }
        converter.setSigningKey(keyValue);

        var accessTokenConverter = new DefaultAccessTokenConverter();
        var userTokenConverter = new DefaultUserAuthenticationConverter();
        userTokenConverter.setUserDetailsService(userDetailsService);
        accessTokenConverter.setUserTokenConverter(userTokenConverter);
        converter.setAccessTokenConverter(accessTokenConverter);
   
        // call afterPropertiesSet() manually since this is not a bean :(
        converter.afterPropertiesSet();

        return converter;
    }
}
```

**Comment by jzheaux** [comment]
I think this sounds reasonable, @tsachev. Would you be interested in submitting a PR to add the annotation and tests?

**Comment by tsachev** [comment]
@jzheaux I created PR  #234 for this.

**Comment by jzheaux** [comment]
Fixed via 5cd1b51f2b49a1f2e472e684ff45eda6ad9e8b1b
